jsonformer/logits_processors.py

```diff
@@ -82,3 +82,54 @@ def __call__(self, _, scores):
         scores[~mask] = -float("inf")
 
         return scores
+
+class IntegerStoppingCriteria(StoppingCriteria):
+    def __init__(
+        self,
+        tokenizer: PreTrainedTokenizer,
+        prompt_length: int,
+        max_digits: int = 15,
+    ):
+        self.tokenizer = tokenizer
+        self.prompt_length = prompt_length
+        self.max_digits = max_digits
+
+    def __call__(
+        self,
+        input_ids: torch.LongTensor,
+        scores: torch.FloatTensor,
+    ) -> bool:
+        decoded = self.tokenizer.decode(
+            input_ids[0][self.prompt_length :], skip_special_tokens=True
+        )
+
+        if len(decoded.strip()) > self.max_digits:
+            return True
+
+        if (
+            len(decoded) > 1
+            and any(c.isdigit() for c in decoded)
+            and decoded[-1] in [" ", "\n"]
```

Comment:
need to check all chars to account for the case where the last sampled token was something like `1<space>h`

Reply:
Shouldn't be necessary since `OutputIntegersTokens` only allows tokens consisting of digits with optional leading and trailing whitespace.

Reply:
Was just doing some more testing with this and am now running into some bugs when these features are used with llama-7b and dolly-v2-12b.